app/addons/activetasks/components.react.jsx

```diff
@@ -0,0 +1,588 @@
+              max="30" 
+              step="1" 
+              value={pollingInterval} 
+              onChange={onChangeHandle}/>
+          </li>
+        </ul>
+      );
+    },
+
+    render: function () {
+      var pollingWidget = this.createPollingWidget();
+
+      return  <div>{pollingWidget}</div>;
+    }
+  });
+
+  var activeTasksHelpers = {
+    getTimeInfo: function (timeStamp) {
+      var timeMessage = [app.helpers.formatDate(timeStamp)];
+      timeMessage.push(app.helpers.getDateFromNow(timeStamp));
+      return timeMessage;
+    },
+
+    getDatabaseFieldMessage: function (item) {
+      var type = item.type;
+      var databaseFieldMessage = [];
+
+      if (type === 'replication') {
+        databaseFieldMessage.push('From: ' + item.source);
+        databaseFieldMessage.push('To: ' + item.target);
+      } else if (type === 'indexer') {
+        databaseFieldMessage.push(item.database);
+        databaseFieldMessage.push('(View: ' + item.design_document + ')');
+      } else {
+        databaseFieldMessage.push(item.database);
+      }
+
+      return databaseFieldMessage;
+    },
+
+    getProgressMessage: function (item) {
+      var progressMessage = [];
+      var type = item.type;
+
+      if (item.hasOwnProperty('progress')) {  
+        progressMessage.push('Progress: ' + item.progress + '%'); 
+      }
+      
+      if (type === 'indexer') { 
+        progressMessage.push(
+          'Processed ' + item.changes_done + ' of ' + item.total_changes + ' changes.'
+        );
+      } else if (type === 'replication') {
+        progressMessage.push(item.docs_written + ' docs written.');
+        
+        if (item.hasOwnProperty('changes_pending')) { 
+          progressMessage.push(item.changes_pending + ' pending changes.'); 
+        }
+      }
+
+      if (item.hasOwnProperty('source_seq')) { 
+        progressMessage.push('Current source sequence: ' + item.source_seq + '. '); 
+      }
+
+      if (item.hasOwnProperty('changes_done')) {
+        progressMessage.push(item.changes_done + ' Changes done.');
+      }
+      
+      return progressMessage;
+    }
+  };
+  
+  return {
+    renderActiveTasks: function (el) {
+      React.render(<ActiveTasksController />, el);
+    },
+
+    removeActiveTasks: function (el) {
+      React.unmountComponentAtNode(el);
+    },
+
+    renderActiveTasksPollingWidget: function (el) {
+      React.render(<ActiveTasksPollingWidget />, el);
+    },
+
+    ActiveTasksController: ActiveTasksController,
+      ActiveTasksFilter: ActiveTasksFilter,
+        ActiveTasksFilterTab: ActiveTasksFilterTab,
+        ActiveTasksFilterTray: ActiveTasksFilterTray,
+
+      ActiveTaskTable: ActiveTaskTable,
+        ActiveTasksTableHeader: ActiveTasksTableHeader,
+        TableHeader: TableHeader,
+        ActiveTasksTableBody: ActiveTasksTableBody,
+        ActiveTaskTableBodyContents: ActiveTaskTableBodyContents,
+
+    ActiveTasksPollingWidget: ActiveTasksPollingWidget
```

Comment:
I was aiming for a kind of hierarchy :( 
but if you think its confusing, I can change it. :)